### PR TITLE
remove keeping_extra because it's replaced with keep_common

### LINKED ISSUE #1656: Remove `keeping_extra`
This was replaced by `keeping_common`

### PR DESCRIPTION
**Pull request by alileza**
Fixes #1656

### CODE REVIEW

File: promql/lex.go [hunk 1]
```diff
@@ -198,21 +198,20 @@ var key = map[string]itemType{
...
-	"offset":        itemOffset,
-	"by":            itemBy,
-	"without":       itemWithout,
-	"keeping_extra": itemKeepCommon,
-	"keep_common":   itemKeepCommon,
-	"on":            itemOn,
-	"ignoring":      itemIgnoring,
-	"group_left":    itemGroupLeft,
-	"group_right":   itemGroupRight,
-	"bool":          itemBool,
```
Comment (juliusv) [inline promql/lex.go]: As discussed in off-PR, let's move the removed keyword into the "Removed keywords" section at the bottom here (and the same in the reverse map!), plus add an if condition in `nextItem()` to print a helpful warning message if someone uses it.

**Comment by juliusv** [comment]
There are some other internal "extra" mentions in the code that would be good to update:

- `keepExtra` -> `keepCommon`
- `KeepExtraLabels` -> `KeepCommonLabels`

File: promql/lex.go (hunk 1 above)
Comment (juliusv) [inline promql/lex.go]: Apologies, there's no "reverse map", I meant the list of tokens in https://github.com/prometheus/prometheus/blob/master/promql/lex.go#L110-L183

**Comment by alileza** [comment]
@juliusv updated !

**Comment by juliusv** [comment]
:+1: pending tests...

**Pull request merged**